python_scripts/cross_validation_sol_02.py

```diff
@@ -135,6 +135,37 @@
 # We see that using `strategy="stratified"`, the results are much worse than
 # with the `most_frequent` strategy. Since the classes are imbalanced,
 # predicting the most frequent involves that we will be right for the
-# proportion of this class (~75% of the samples). However, by using the
-# `stratified` strategy, wrong predictions will be made even for the most
+# proportion of this class (~75% of the samples). However, the `stratified`
+# strategy will randomly generate predictions by respecting the training
+# set’s class distribution, resulting in wrong predictions even for the most
 # frequent class, hence we obtain a lower accuracy.
+#
+# Notice that the `most_frequent` strategy is deterministic for a given
+# cross-validation split, whereas the `stratified` strategy will lead
+# to different results even for the same cv split due to the extra randomization.
```

Comment:
It might be worth to add "... if one does not set the parameter `random_state`."